app/src/main/java/com/connorlay/knitgrid/fragments/BasePatternFragment.java

```diff
@@ -130,4 +158,25 @@ protected void setGridBackgroundColor(int color) {
         }
     }
 
+    protected void setGridBackgroundMultiColor(){
+        List<StitchPatternRelation> list = mPatternPresenter.getPattern().getStitchRelations();
+        for (StitchPatternRelation s: list) {
+            int oldColor = s.getColorID();
+            int newColor = mPatternPresenter.getStitch(s.getRow(), s.getCol()).getColorID();
+            if (oldColor != newColor && newColor != mCellDefaultColor && oldColor != mCellDefaultColor) {
```

Comment:
Can you wrap these boolean expression in named private methods to make it clearer what the different cases are here?